subsys/net/ip/tcp2.c

```diff
@@ -391,8 +391,8 @@ static bool tcp_options_check(void *buf, ssize_t len)
 			continue;
 		} else {
 			if (len < 2) { /* Only END and NOP can have length 1 */
-				NET_ERR("Illegal option %d with length %d",
-					opt, len);
+				NET_ERR("Illegal option %d with length %ld",
```

Comment:
Instead of the cast, why not use "%zd" as a modifier. Also fix the commit body to reflect that.

Reply:
done.